src/java_bytecode/java_static_initializers.cpp

```diff
@@ -134,82 +260,239 @@ static void create_clinit_wrapper_symbols(
     "clinit wrapper");
 }
 
-/// Produces the static initialiser wrapper body for the given function.
+/// Produces the static initialiser wrapper body for the given function. This
+/// static initializer implements (a simplification of) the algorithm defined
+/// in Section 5.5 of the JVM Specs. This function, or wrapper, checks whether
+/// static init has already taken place, calls the actual `<clinit>` method if
+/// not, and possibly recursively initializes super-classes and interfaces.
+///
+/// ```
+///   bool was_ready;
+///   if(java::C::__CPROVER_clinit_local_state == RETURN)
+///   {
+///     return;
+///   }
+///   java::C::__CPROVER_clinit_local_state = RETURN;
+///
+///   ATOMIC_BEGIN
+///   Assume(java::C::__CPROVER_clinit_state != IN_PROGRESS)
+///   If(java::C::__CPROVER_clinit_state == NOT_INIT)
+///   {
+///     java::C::__CPROVER_clinit_state = IN_PROGRESS
+///     was_ready = false;
+///   }
+///   else If(java::C::__CPROVER_clinit_state == READY)
```

Comment:
Out of date? AFAIK there is no state named READY anymore

Reply:
yep this should be RETURN